src/sinks/datadog/metrics.rs

```diff
@@ -152,33 +194,37 @@ impl SinkConfig for DatadogConfig {
     }
 }
 
-#[async_trait::async_trait]
-impl HttpSink for DatadogSink {
-    type Input = Event;
-    type Output = Vec<Metric>;
-
-    fn encode_event(&self, event: Event) -> Option<Self::Input> {
-        Some(event)
-    }
-
-    async fn build_request(&self, events: Self::Output) -> crate::Result<Request<Vec<u8>>> {
+impl DatadogSink {
+    fn build_request(
+        &self,
+        events: PartitionInnerBuffer<Vec<Metric>, DatadogEndpoint>,
+    ) -> crate::Result<Request<Vec<u8>>> {
+        let (events, ep) = events.into_parts();
         let now = Utc::now().timestamp();
-        let interval = now - self.last_sent_timestamp.load(SeqCst);
-        self.last_sent_timestamp.store(now, SeqCst);
+        let interval = now - self.last_sent_timestamp[ep as usize].load(SeqCst);
+        self.last_sent_timestamp[ep as usize].store(now, SeqCst);
 
-        let input = encode_events(events, interval, &self.config.namespace);
-        let body = serde_json::to_vec(&input).unwrap();
+        let body = match ep {
+            DatadogEndpoint::Series => {
+                let input = encode_events(events, interval, &self.config.namespace);
+                serde_json::to_vec(&input).unwrap()
+            }
+            DatadogEndpoint::Distribution => {
+                let input = encode_distribution_events(events, interval, &self.config.namespace);
+                serde_json::to_vec(&input).unwrap()
```

Comment:
nitpick: We could pull this out of the `match` since it's the same for both.

Reply:
`encode_distribution_events` and `encode_events` return different types, only when we serialize them they become the same.